Changes required to migrate from uTorrent.

zoomorph opened the pull request:
Just submitting some changes that I made in migrating from uTorrent, in case it helps others.

Commit:
Changes required to migrate from uTorrent.

mikedld commented:
Thanks. Could you provide some more details as to why each of the changes was necessary? Also also if possible, could you attach (or send me via email, if you don't wish to make it public) your state directory so that I'm able to reproduce? What version of uTorrent are you using?

zoomorph commented:
Sorry, I didn't do clean commits, I just hacked it together until it worked and I'm not too familiar with any of this stuff so my solution may not be the most correct. In fact there were 5-10 torrents out of a few thousand that I just fixed manually at the end.

I think my main issue was that I renamed the top level directory torrents were saved into. With Transmission 2.94, it seems to add "files" key to the resume file when this is done. Just doing this wasn't enough. The torrent and resume files also seemingly needed to be named in a way that matched the original torrent file name (which I got from "caption").

DoNotDownload was not working by default, coming from uTorrent 2.1.x on Windows (quite and old one).

Catching the boost filesystem exception to work around issue#19.

I think that's it.

PF4Public commented:
I can confirm, this worked for uTorrent 1.8. Most of the files were OK, but several needed to be fixed manually afterwards.

Neustradamus commented:
Any news on it?

PF4Public commented:
@Neustradamus What news do you expect? :) Just build from this PR and be done with it :)

Neustradamus commented:
It has not been merged yet since 2020-10-15.
